enhancements/network/communication-ingress-flows-matrix.md

```diff
@@ -0,0 +1,198 @@
+from the existing `EndpointSlices` in the OpenShift cluster.
+
+`EndpointSlices` group network endpoints together by unique combinations of 
+protocol, port number, and `Service` name and therefore can be used as the 
+source of truth for cluster traffic. For that to work, every application that 
+attracts ingress traffic running in the cluster should have a `Service` object,
+the control plane automatically creates `EndpointSlices` for any `Service` that 
+has a selector specified.
+
+To generate the communication matrix the `EndpointSlices` in the cluster should 
+be filtered by `Service` type `NodePort` and `LoadBalancer` as there are also 
+`EndpointSlices` in the cluster for internal `Services`. The node hosted services 
+such as sshd, rpc, etc. are currently missing `EndpointSlices` and should be created 
+by the Machine API Operator during cluster deployment.
+
+Sabina to Yuval: What k8s link you wanted to add here?
+
+The module allows to generate the communication matrix in various formats:
+- YAML/JSON
+- CSV
+- nftables
+
+Each record describes a flow with the following information:
+```
+traffic-direction     Data flow direction (ingress/egress)
+protocol              Network protocol (TCP/UDP)
+port-number           Flow port number
+service-name          OpenShift Service name
+namespace             Service namespace
+node-role             Service node host role (control-plane/master/worker)
+optional              Optional or mandatory flow for OpenShift
+```
+
+For example,
+`ingress,TCP,18080,openshift-kni-infra-coredns-cnfdr8.telco5g.eng.rdu2.redhat.com-coredns,worker,true`
+
+### Risks and Mitigations
+
+For egress flows support (out of scope) an API change might be needed. 
+`EndpointSlices` stand better for ingress traffic than for egress, in ingress traffic the 
+services are less dynamic and mostly stay up during the entire cluster lifetime.
+Supporting egress traffic might require changes in the API that should be reviewed and 
+agreed with the upstream Kubernetes community.
+
+### Drawbacks #Sabina: anything relevant here?
+
+The idea is to find the best form of an argument why this enhancement should
+_not_ be implemented.  
+
+What trade-offs (technical/efficiency cost, user experience, flexibility, 
+supportability, etc) must be made in order to implement this? What are the reasons
+we might not want to undertake this proposal, and how do we overcome them?  
+
+Does this proposal implement a behavior that's new/unique/novel? Is it poorly
+aligned with existing user expectations?  Will it be a significant maintenance
+burden?  Is it likely to be superceded by something else in the near future?
+
+## Design Details
+
+### Open Questions
+
+1. Is there a better OpenShift object to use than `EndpointSlices` to collect 
+   information about ingress traffic?
+
+2. Should a new OpenShift object be introduced to act as the source of truth
+   for cluster traffic?
+
```

Comment:
3. how to deal with privlieged (hostNet: true) workloads that can open new ports?
4. how to describe host services (probably by integrating with CAPI)
5. do we want to include both tools to get the declared flows and the actual currently running ones? maybe even offer a validation mode that will issue alerts on any miss. I'd also mention this whole thing might evolve into it's own operator